### PR TITLE
test: official site checker before publish

### PR DESCRIPTION
**Pull request by ycjcl868**
image

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [x] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: scripts/check-site.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+function checkKeyFilesExist() {
+  const keyfiles = [
+    '/index.html',
+    '/index-cn.html',
+    '/404.html',
+    '/CNAME',
+    '/color.less',
+    '/config.yml',
+    '/changelog.html',
+    '/changelog-cn.html',
```
Comment (afc163) [inline scripts/check-site.js]: 所有 components 的 html 也都可以检查一遍。

**Comment by ycjcl868** [comment]
Close https://github.com/benjycui/bisheng/pull/209.

File: scripts/check-site.js (hunk 1 above)
Comment (ycjcl868) [reply scripts/check-site.js]: 要从 github 上获取样本做对比

**Comment by afc163** [comment]
还是可以加，双重保险。

**Comment by ycjcl868** [comment]
变更文件名或者新增文件，做 diff 会比较麻烦。

**Comment by ycjcl868** [comment]
改成 jest ，发布前跑

File: scripts/check-site.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  return filename
+    .replace(/(?<=\.)\w{6}(?:\.)/g, '')
+    .replace(/(?<=demo\-0\.).*?(?:\.)/g, '')
+    .replace(/(?<=\-)\w{8}(?:-)/g, '')
+    .replace(/(?<=\-)\w{8}(?:\.)/g, '');
+};
+
+const getPrevfiles = async () => {
+  const { tree } = await fetch(
+    'https://api.github.com/repos/ant-design/ant-design/git/trees/gh-pages?recursive=1',
```
Comment (afc163) [inline scripts/check-site.js]: 容易被 ip 限制跑挂。https://github.com/ant-design/ant-design/blob/f2e541ade5ad62e74842202c3473165363a24e6e/index.js#L7

File: scripts/check-site.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+const cheerio = require('cheerio');
+const glob = require('glob');
+const uniq = require('lodash/uniq');
+const { createServer } = require('http-server');
+const zhCN = require('../site/theme/zh-CN');
+const enUS = require('../site/theme/en-US');
+
+// const componentRouteMap = {
+//   '/components/col':
+// }
```
Comment (afc163) [inline scripts/check-site.js]: 可以删掉。

Comment (ycjcl868) [reply scripts/check-site.js]: OK

**Pull request merged**